src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -701,6 +701,13 @@ public override BoundNode VisitArrayAccess(BoundArrayAccess node)
         internal static bool IsFieldOrPropertyInitializer(BoundStatement initializer)
         {
             var syntax = initializer.Syntax;
+
+            if (syntax.IsKind(SyntaxKind.Parameter))
```

Comment:
nit: by the looks of this and previous code, it seems we are missing information in bound nodes (we have to go back to syntax). Consider filing an issue.